src/views/mail.py

```diff
@@ -0,0 +1,98 @@
+# Olivier Dion - 2019
+
+from src.models.mail   import *
+from src.views.widgets import (
+    Buffer,
+    Button,
+    Controller,
+    Signal,
+    TreeWidget,
+)
+
+from src.views.tui     import TUI
+
+from urwid import (
+    AttrMap,
+    Columns,
+    Filler,
+    LineBox,
+    Pile,
+    ProgressBar,
+)
+
+@Signal("on_quit")
+class MailView(Controller):
+
+    def __init__(self):
+
+        super().__init__()
+
+        subject  = LineBox(Buffer(("header", "Subject\n\n"), f"{default_subject}"))
+
+        sender   = LineBox(Buffer(("header", "Sender\n\n"), ""))
+        receiver = LineBox(Buffer(("header", "Receiver\n\n"), f"{debug_receiver}"))
+
+        infos    = Columns([sender, receiver])
+
+        message  = LineBox(Buffer(("header" ,"Message\n\n"), "", multiline=True))
+
+        confirm  = LineBox(AttrMap(Button("Confirm",
+                                          on_press=self.confirm,
+                                          align="center"),
+                                   "default",
+                                   "confirm_button"))
+
+        abort    = LineBox(AttrMap(Button("Abort",
+                                          on_press=self.abort,
+                                          align="center"),
+                                   "default",
+                                   "abort_button"))
+
+        buttons  = Columns([confirm, abort])
+
+        tree     = TreeWidget(subject)
+
+        tree.split_vertically(infos)
+        tree.split_vertically(message)
+        tree.split_vertically(buttons)
+
+        tree.bind("up",  tree.focus_prev_node)
+        tree.bind("down",tree.focus_next_node)
+
+        tree.set_aliases(
+            [
+                ("ctrl p",    "up"),
+                ("shift tab", "up"),
+                ("ctrl n",    "down"),
+                ("tab",       "down"),
+                ("ctrl f",    "right"),
+                ("ctrl b",    "left"),
+            ]
+        )
+
+        bar = ProgressBar("default", "confirm_button", current=50)
+
+        self.root     = Pile([Filler(tree), Filler(bar)])
+        self.bar      = bar
+        self.subject  = subject
+        self.sender   = sender
+        self.receiver = receiver
+        self.message  = message
+
+    # Text from decorator, cuz im lazy
+    @staticmethod
+    def tfd(d):
```

Comment:
imo tu devrais en faire une extension method:
https://stackoverflow.com/questions/514068/extension-methods-in-python